pkg/api/types.go

```diff
@@ -715,6 +717,17 @@ func (p *Properties) HasVirtualMachineScaleSets() bool {
 	return false
 }
 
+// HasAllZonesAgentPools will return true if all of the agent pools have zones
```

Comment:
Let's go full java and change this func name to `HasZonesForAllAgentPools`

Reply:
:'(